tests/FastJsonPatchTest.php

```diff
@@ -177,6 +193,43 @@ public function testValidJsonPatches(string $json, string $patches, string $expe
         );
     }
 
+    /**
+     * @return array<string, string[]>
+     */
+    public static function validOperationsForDTOsProvider(): iterable
+    {
+        foreach (self::validOperationsProvider() as $name => $values) {
+            if ($name === 'TestHandler with optional patch properties') {
+                // DTOs do not support arbitrary constructor parameters
+                continue;
+            }
+
+            yield $name => $values;
+        }
+    }
+
+    /**
+     * @param string $json
+     * @param string $patches
+     * @param string $expected
+     * @return void
+     * @throws \JsonException
+     * @throws FastJsonPatchException
+     */
+    #[DataProvider('validOperationsForDTOsProvider')]
+    public function testValidJsonPatchesAsDTOs(string $json, string $patches, string $expected): void
```

Comment:
Potentially this test runs more cases than is strictly necessary to prove the DTOs are correctly read & handled.

However it seemed the most efficient way to have confidence that they support all the expected properties / types and work correctly with all handlers and to keep that in sync with any changes to the pure JSON objects over time.